### DISCUSSION TITLE
Converting JPG to PDF - Hundreds of Folders

### DISCUSSION THREAD

**Discussion by kstarrox128** [ImageMagick/ImageMagick#2009] [General]
Hey,
Ive got hundreds of folders with JPGs in them which id like to combine to PDF in their current state, so keeping the images in their respective folder but also combine them to PDF thats named after the folder.
Ive seen some other posts about it online and people have solutions, but im using Windows not mac or linux and am struggling to figure out the command lines i would use. This example [Here](http://www.imagemagick.org/discourse-server/viewtopic.php?t=36283)  explains a script has been written, but im trying to understand if the script is some sort of bat file or if its done another way - i would comment on that post ive linked but whwn i click register it takes me here and then doesnt let me log in.
If someone is able to help with what the windows command lines would be or how to do the script would be great.
Thanks!

**Comment by snibgo** [comment]
Yes, those are BAT scripts.

**Reply by kstarrox128** [reply]
How would the entire BAT file be written in that case, this is the main part im unsure of? Does a BAT file need to be stored in a parent folder or something - this is where i need guidance.

**Comment by snibgo** [comment]
I'm not sure what guidance I can sensibly offer. I suggest you find a tutorial on Windows BAT scripts.

**Comment by minden-xyz** [comment]
Hi, 
I'm also trying to do this same thing. On Mac.
How can i run a command that combines all jpgs inside each folder into a PDF and renames the PDF to the name of the folder it resides in.
So in the example below, Im wanting to create 3 PDFS (Folder A, Folder B and Folder C) by right clicking and running batch on the parent folder.

Example:
- Parent folder 
**|- Folder A**
||- test1.jpg
||- test2.jpg
||- test3.jpg
**|- Folder B**
||- example1.jpg
|| - example2.jpg
**|- Folder C**
||- temp.jpg
||- temp2.jpg

So there would be a pdf called Folder A.pdf, in Folder A. Then a pdf called Folder B.pdf, in Folder B. And so on. 

I'm fairly new to Terminal and IM, so if you could provide step by step inputs that would be awesome, thanks in advance.

**Comment by Jmuccigr** [comment]
The overall approach to these problems requires some minor scripting, but can I suggest that people have a look at img2pdf for the actual conversion work? It will neatly embed an image in a pdf and will also produce PDFs of a determined size, so you could, for example, convert a folder full of images to a PDF that is A4 sized. (It will take multiple image files as input and produce one PDF as output.) The images aren't re-encoded, they're simply embedded in the PDF.

IM is great, but I'd suggest this isn't a good use-case for it.

https://sourceforge.net/projects/img2pdf/

PS Mac users can install via homebrew.
